frontend/src/lib/StaticGraphParser.ts

```diff
@@ -288,10 +288,14 @@ export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
  *
  * @param graph The dagre graph object
  */
-export function transitiveReduction(graph: dagre.graphlib.Graph): dagre.graphlib.Graph | undefined {
-  // safeguard against too big graphs
-  if (!graph || graph.edgeCount() > 1000 || graph.nodeCount() > 1000) {
-    return undefined;
+export function transitiveReduction(graph: dagre.graphlib.Graph): dagre.graphlib.Graph {
+  // Safeguard against excessively large graphs
+  if (!graph) {
+    throw new Error("Graph object doesn't exist.");
+  } else if (graph.nodeCount() > 1000) {
+    throw new Error('Graph node count exceeds 1000.');
+  } else if (graph.edgeCount() > 1000) {
+    throw new Error('Graph edge count exceeds 1000.');
   }
```

Comment:
Have you tested these two scenarios locally?

Reply:
Not yet. Can do though. Hopefully it doesn't clobber my local K8s, heh.

Reply:
Hey, @zijianjoy. I tested this locally. It seems to be working as expected. The only problem is that trying to spawn a pipeline with 1000 components crashed my local k8s cluster. I temporarily reduced the condition from 1000 to 4 for validation purposes. When I submit the following pipeline:

```python
def ran() -> None:
    from random import random
    print(random(1, 1000))

ran_component = kfp.components.create_component_from_func(ran)

@kfp.dsl.pipeline(name="thousand")
def large_pipeline():
    step = ran_component()
    step = ran_component()
    step = ran_component()
    step = ran_component()
```

I get the following result in the GUI:
![](https://i.imgur.com/pA3GbgG.jpg)

Reply:
Thank you for the validation. I was under the impression that frontend shouldn't load the large graph that exceeded threshold. But looks like it is still loading the large graph with your change. Can we avoid loading the large graph?